src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -66,43 +66,57 @@ leader_balancer::leader_balancer(
     }
 }
 
+void leader_balancer::on_leadership_change(
+  raft::group_id group, model::term_id, std::optional<model::node_id>) {
+    // Don't bother doing anything if it's not enabled
+    if (!_enabled()) {
+        return;
+    }
+
+    if (group == _raft0->group()) {
+        // Active leader balancer again if leadership of
+        // raft0 is transfered to this node.
+        if (_raft0->is_elected_leader()) {
+            vlog(
+              clusterlog.info,
+              "Leader balancer: controller leadership detected. "
+              "Starting "
+              "rebalancer in {} seconds",
+              std::chrono::duration_cast<std::chrono::seconds>(
+                leader_activation_delay)
+                .count());
+
+            _timer.cancel();
+            _timer.arm(leader_activation_delay);
+        } else {
+            vlog(
+              clusterlog.info,
+              "Leader balancer: node is not controller leader");
+            _need_controller_refresh = true;
+            _timer.cancel();
+            _timer.arm(_idle_timeout());
+        }
+    }
+
+    if (!_raft0->is_elected_leader()) {
+        return;
+    }
+
+    // Update in flight state
+    if (auto it = _in_flight_changes.find(group);
```

Comment:
I think this block is the only changed part of the function, right?

In future please do non-functional refactors in a separate commit to functional changes (i.e. in this case it would be one commit to pull out the implementation to the `on_leadership_change` function, then a separate commit that adds the new logic to the end of the function -- much easier to review that way.

Reply:
Will keep this in mind. And yes that's the only changed part of the function.